scripts/build-renode.sh

```diff
@@ -0,0 +1,162 @@
+eval $(make env)
+make info
+
+source $SCRIPT_DIR/settings.sh
+
+set -x
+set -e
+
+RENODE_BIN=${RENODE_BIN:-renode}
+if ! command -v $RENODE_BIN 2>&1 1>/dev/null; then
+	# Download prebuilt renode Release if none is currently installed
+
+	RENODE_PACKAGE=renode-${RENODE_VERSION}-1-x86_64.pkg.tar.xz
+	RENODE_URL=https://github.com/renode/renode/releases/download/v$RENODE_VERSION/$RENODE_PACKAGE
+	RENODE_LOCATION="$BUILD_DIR/renode"
+	RENODE_BIN=$RENODE_LOCATION/opt/renode/bin/Renode.exe
+
+	if [ ! -x $RENODE_BIN ]; then
+		mkdir -p $RENODE_LOCATION
+		(
+			cd $RENODE_LOCATION
+			wget $RENODE_URL
+			tar -xf $RENODE_PACKAGE
+		)
+		
+		chmod u+x $RENODE_BIN
+		echo "Renode downloaded and installed locally: $RENODE_BIN" 
+	fi
+fi
+
+case $CPU in
+	vexriscv | picorv32)
+		;;
+	*)
+		echo "CPU $CPU_TYPE isn't supported at the moment."
+		exit 1
+		;;
+esac
+
+LITEX_CONFIG_FILE="$TARGET_BUILD_DIR/test/csr.csv"
+if [ ! -f $LITEX_CONFIG_FILE ]; then
+	make firmware
+fi
+
+function parse_generated_header {
+	# $1 - filename
+	# $2 - define name
+
+	grep $2 $TARGET_BUILD_DIR/software/include/generated/$1 | tr -s ' ' | cut -d' ' -f3
+}
+ETH_BASE_ADDRESS=$(parse_generated_header "csr.h" CSR_ETHMAC_BASE)
+
+# Ethernet
+if [ ! -z "$ETH_BASE_ADDRESS" ]; then
+	RENODE_NETWORK=${RENODE_NETWORK:-tap}
+	case $RENODE_NETWORK in
+	tap)
+		echo "Using tun device for Renode networking, (may need sudo)..."
+		# Make the tap0 dev node exists
+		if [ ! -e /dev/net/tap0 ]; then
+			sudo true
+			sudo mknod /dev/net/tap0 c 10 200
+			sudo chown $(whoami) /dev/net/tap0
+		fi
+
+		# Check that the tap0 network interface exists
+		if [ ! -e /sys/class/net/tap0 ]; then
+			sudo true
+			if sudo which openvpn > /dev/null; then
+				sudo openvpn --mktun --dev tap0 --user $(whoami)
+			elif sudo which tunctl > /dev/null; then
+				sudo tunctl -t tap0 -u $(whoami)
+			else
+				echo "Unable to find tool to create tap0 device!"
+				exit 1
+			fi
+		fi
+
+		# Check the tap0 device if configure and up
+		if sudo which ifconfig > /dev/null; then
+			if ! ifconfig tap0 | grep -q "UP" || ! ifconfig tap0 | grep -q "$TFTP_IPRANGE.100"; then
+				sudo true
+				sudo ifconfig tap0 $TFTP_IPRANGE.100 netmask 255.255.255.0 up
+			fi
+		elif sudo which ip > /dev/null; then
+			if ! ip addr show tap0 | grep -q "UP" || ! ip addr show tap0 | grep -q "$TFTP_IPRANGE.100"; then
+				sudo true
+				sudo ip addr add $TFTP_IPRANGE.100/24 dev tap0
+				sudo ip link set dev tap0 up
+			fi
+		else
+			echo "Unable to find tool to configure tap0 address"
+			exit 1
+		fi
+
+		# Build/copy the image into the TFTP directory.
+		make tftp
+
+		# Restart tftpd
+		make tftpd_stop
+		make tftpd_start
+
+		TAP_INTERFACE=tap0
+		;;
+
+	none)
+		echo "Renode networking disabled..."
+		;;
+	*)
+		echo "Unknown RENODE_NETWORK mode '$RENODE_NETWORK'"
+		return 1
+		;;
+	esac
+fi
+
+script_file=$(mktemp /tmp/litex_buildenv.resc.XXXXXX)
+platform_file=$(mktemp /tmp/litex_buildenv.repl.XXXXXX)
```

Comment:
Lets put these files in the build directory under `build/<build directory>/renode`?

Reply:
We then don't need to clean them up and people can launch renode manually with the configurations.

Reply:
This is a good idea.

Reply:
It's done.